app/assets/javascripts/specifics.js.coffee

```diff
@@ -564,7 +569,23 @@ class Row
   toHashKey: ->
     @flattenToIds().join(",") + ",#{@specificLeftMost}" + ",#{@nonSpecificLeftMost?.id}"
 
-  
+  # If the row references a leftmost, either specific or not, return the event(s)
+  # (because a UNION can place multiple events in the specific leftMost, this can be > 1)
+  leftMostEvents: ->
+    if @nonSpecificLeftMost?
+      return [@nonSpecificLeftMost]
+    if @specificLeftMost? && _.isString(@specificLeftMost)
+      specificIndex = hqmf.SpecificsManager.getColumnIndex(@specificLeftMost)
+      return [@values[specificIndex]] if @values[specificIndex]? && @values[specificIndex] != hqmf.SpecificsManager.any
+    if @specificLeftMost? && _.isObject(@specificLeftMost)
+      events = []
```

Comment:
is there any possibility of a repeated event? In the previous leftmost code, there was a uniqueness filter being applied.